packages/cli/cloud/src/services/logger.ts

```diff
@@ -0,0 +1,158 @@
+import chalk from 'chalk';
+import ora from 'ora';
+import * as cliProgress from 'cli-progress';
+
+export interface LoggerOptions {
+  silent?: boolean;
+  debug?: boolean;
+  timestamp?: boolean;
+}
+
+export interface Logger {
+  warnings: number;
+  errors: number;
+  debug: (...args: unknown[]) => void;
+  info: (...args: unknown[]) => void;
+  success: (...args: unknown[]) => void;
+  warn: (...args: unknown[]) => void;
+  error: (...args: unknown[]) => void;
+  log: (...args: unknown[]) => void;
+  spinner: (text: string) => Pick<ora.Ora, 'succeed' | 'fail' | 'start' | 'text'>;
+  progressBar: (
+    totalSize: number,
+    text: string
+  ) => Pick<cliProgress.SingleBar, 'start' | 'stop' | 'update'>;
+}
+
+const createLogger = (options: LoggerOptions = {}): Logger => {
+  const { silent = false, debug = false, timestamp = true } = options;
+
+  const state = { errors: 0, warning: 0 };
+
+  return {
+    get warnings() {
+      return state.warning;
+    },
+
+    get errors() {
+      return state.errors;
+    },
+
+    debug(...args) {
+      if (silent || !debug) {
+        return;
+      }
+
+      console.log(
+        chalk.cyan(`[DEBUG]${timestamp ? `\t[${new Date().toISOString()}]` : ''}`),
+        ...args
+      );
+    },
+
+    info(...args) {
+      if (silent) {
+        return;
+      }
+
+      console.info(
+        chalk.blue(`[INFO]${timestamp ? `\t[${new Date().toISOString()}]` : ''}`),
+        ...args
+      );
+    },
+
+    log(...args) {
+      if (silent) {
+        return;
+      }
+
+      console.info(chalk.blue(`${timestamp ? `\t[${new Date().toISOString()}]` : ''}`), ...args);
+    },
+
+    success(...args) {
+      if (silent) {
+        return;
+      }
+
+      console.info(
+        chalk.green(`[SUCCESS]${timestamp ? `\t[${new Date().toISOString()}]` : ''}`),
+        ...args
+      );
+    },
+
+    warn(...args) {
+      state.warning += 1;
+
+      if (silent) {
+        return;
+      }
+
+      console.warn(
+        chalk.yellow(`[WARN]${timestamp ? `\t[${new Date().toISOString()}]` : ''}`),
+        ...args
+      );
+    },
+
+    error(...args) {
+      state.errors += 1;
+
+      if (silent) {
+        return;
+      }
+
+      console.error(
+        chalk.red(`[ERROR]${timestamp ? `\t[${new Date().toISOString()}]` : ''}`),
+        ...args
+      );
+    },
+
+    // @ts-expect-error – returning a subpart of ora is fine because the types tell us what is what.
+    spinner(text: string) {
+      if (silent) {
+        return {
```

Comment:
you could make all those default consts to reuse the same obj and declare once

Reply:
Done